### PR TITLE
Fix bad reference in node.rb (fixes #2398)

### LINKED ISSUE #2398: Fix bad reference in node.rb
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

This issue is to fix a bad reference to `node` which should have been `self` -- to fix the display of http://publiclab.org/methods

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [app/models/node.rb](https://github.com/publiclab/plots2/blob/master/node.rb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://publiclab.github.io/community-toolbox/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -723,7 +723,7 @@ def body_preview(length = 100)
   def questions
     # override with a tag like `questions:h2s`
     if self.has_power_tag('questions')
-      tagname = node.power_tag('questions')
+      tagname = self.power_tag('questions')
     else
       tagname = self.slug_from_path
     end
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

(This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).)

### PR DESCRIPTION
**Pull request by MittalS211**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

Fixes #2398

### CODE REVIEW

**Comment by MittalS211** [comment]
@jywarren  hi I just submitted the PR for #2398 
Thanks for the help, I shall now move on to a new issue.

File: app/models/node.rb [hunk 1]
```diff
@@ -713,6 +1217,373 @@ def body_preview(length = 100)
     try(:latest).body_preview(length)
   end
 
+  # so we can quickly fetch questions corresponding to this node
+  # with node.questions
+  def questions
+    # override with a tag like `questions:h2s`
+    if self.has_power_tag('questions')
+      tagname = self.power_tag('questions')
```
Comment (seafr) [inline app/models/node.rb]: This is the only line that needed to be changed.

**Comment by seafr** [comment]
Thank you @MittalS211 for your first PR.
Issue #2398 specified **only** one line to be changed in `app/models/node.rb`. This is probably why the Travis CI build was not successful (see the red `X` above).
Have you run the tests locally?

**Comment by jywarren** [comment]
This may help with removing unneeded changes!

https://stackoverflow.com/questions/215718/reset-or-revert-a-specific-file-to-a-specific-revision-using-git

**Comment by MittalS211** [comment]
Alright. I'll just go over it and update you.

**Comment by seafr** [comment]
@MittalS211 I would like to ask if you are working on a `clone` of `plots2`, or from a downloaded zip archive (since it seems that entire files were added to the commits).

File: app/views/notes/_comment.html.erb [hunk 2]
```diff
@@ -44,6 +44,9 @@
           <i class='icon fa fa-trash'></i>
         </a>
       <% end %>
+          <a rel="tooltip" title="Flag as spam." class="btn btn-sm btn-default" href="mailto:moderators@publiclab.org?subject=Reporting+spam+on+Public Lab&body=Hi,+I+found+this+item+that+looks+like+spam+or+needs+to+be+moderated:+"<%= comment.body.replace(' ','+') %>"%0A%0Ahttps://publiclab.org/n/<%= node.id %>%0Aby+https://publiclab.org/profile/<%= comment.author.username %>%0A%0AThanks!">
```
Comment (jywarren) [inline app/views/notes/_comment.html.erb]: These lines look just right. I wonder if you could remove the others from your PR? 

https://stackoverflow.com/questions/215718/reset-or-revert-a-specific-file-to-a-specific-revision-using-git

Then I think we'll be done here! Thanks so much!

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR.
Thanks for contributing at Public Lab

**Pull request closed without merging**